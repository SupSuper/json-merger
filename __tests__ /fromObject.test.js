const helpers = require("../__helpers__");

describe(".fromObject() should process the object and return", function () {

    test("the processed object if config.indicatorPrefix is @", function () {

        const object = {
            "a": {
                "@value": 1,
            },
            "$value": 2
        };

        const result = helpers.fromObject(object, {
            indicatorPrefix: "@"
        });

        expect(result).toMatchSnapshot();
    });

    test("the processed object if config.stringify is false", function () {

        const object = {
            "a": {
                "$value": 1
            }
        };

        const result = helpers.fromObject(object, {
            stringify: false
        });

        expect(result).toMatchSnapshot();
    });

    test("a JSON string if config.stringify is true", function () {

        const object = {
            "a": {
                "$value": 1
            }
        };

        const result = helpers.fromObject(object, {
            stringify: true
        });

        expect(result).toMatchSnapshot();
    });

    test("a pretty JSON string if config.stringify is 'pretty'", function () {

        const object = {
            "a": {
                "$value": 1
            }
        };

        const result = helpers.fromObject(object, {
            stringify: "pretty"
        });

        expect(result).toMatchSnapshot();
    });
});
