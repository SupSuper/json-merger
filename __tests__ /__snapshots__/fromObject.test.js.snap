// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.fromObject() should process the object and return a JSON string if config.stringify is true 1`] = `"{\\"a\\":1}"`;

exports[`.fromObject() should process the object and return a pretty JSON string if config.stringify is 'pretty' 1`] = `
"{
	\\"a\\": 1
}"
`;

exports[`.fromObject() should process the object and return the processed object if config.indicatorPrefix is @ 1`] = `
"{
	\\"a\\": 1,
	\\"$value\\": 2
}"
`;

exports[`.fromObject() should process the object and return the processed object if config.stringify is false 1`] = `
Object {
  "a": 1,
}
`;
